src/gui/uithememanager.h

```diff
@@ -48,13 +48,20 @@ class UIThemeManager : public QObject
     QIcon getIcon(const QString &iconId) const;
     QIcon getIcon(const QString &iconId, const QString &fallback) const;
     QIcon getFlagIcon(const QString &countryIsoCode) const;
+    QPixmap getScaledPixmap(const QString &iconId, const QWidget *widget, int baseHeight) const;
+    QString getIconPath(const QString &iconId) const;
 
 private:
-    UIThemeManager(); // singleton class
-    QString getIconPath(const QString &iconId) const;
+    using IconMap = QHash<QString, QString>;
+
+    static bool loadIconConfig(const QString &configFile, const QString &iconDir, IconMap &config);
```

Comment:
I would make it non-static and drop last parameter.

Reply:
On the other hand, it should be really class-independent (and it is actually used to load different files). Then just put it in an anonymous namespace in .cpp. You will also have to change the error handling in it (throw Runtime Error in it and print log message in calling code).

IMO, using static methods in C++ is unjustified in most cases.